src/plugins/csp_handler/server/plugin.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  CoreSetup,
+  CoreStart,
+  Logger,
+  Plugin,
+  PluginInitializerContext,
+} from '../../../core/server';
+
+import { createCspRulesPreResponseHandler } from './csp_handlers';
+import { AppPluginSetupDependencies, CspHandlerPluginSetup, CspHandlerPluginStart } from './types';
+
+export class CspHandlerPlugin implements Plugin<CspHandlerPluginSetup, CspHandlerPluginStart> {
+  private readonly logger: Logger;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+  }
+
+  public async setup(core: CoreSetup, { applicationConfig }: AppPluginSetupDependencies) {
+    core.http.registerOnPreResponse(
+      createCspRulesPreResponseHandler(
+        core,
+        core.http.csp.header,
+        applicationConfig.getConfigurationClient,
```

Comment:
Can we pass Client here, instead of passing function?

Reply:
Replied as part of https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5641/files#r1518231638

Reply:
Can't we `getConfigurationClient(coreStart.opensearch.client.asScoped(request)` call this before` core.http.registerOnPreResponse()`?

Reply:
This block of code is in `setup`. There is no access to `coreStart` yet. Furthermore, there is no `request` coming in yet. The code is written in this way so that we can leverage the request context inside the request handler.

Reply:
Got it.